File: sdks/nodejs/src/providers/meta.ts
import type { CyberArmorClient } from '../client';
import { CyberArmorOpenAICompatible } from './openaiCompatible';

export class CyberArmorMeta extends CyberArmorOpenAICompatible {
  constructor(ca: CyberArmorClient, options?: Record<string, unknown>) {
    super(ca, 'meta', options, 'llama-3.3-70b-instruct');
  }
}

export function protectedMetaClient(ca: CyberArmorClient, options?: Record<string, unknown>): CyberArmorMeta {
  return new CyberArmorMeta(ca, options);
}

File: sdks/rust/src/providers/meta.rs
use std::sync::Arc;
use crate::client::CyberArmorClient;
use super::openai_compatible::OpenAICompatibleProvider;

/// Builds a Meta provider wrapper backed by an OpenAI-compatible endpoint.
pub fn new(client: Arc<CyberArmorClient>, api_key: impl Into<String>, tenant_id: impl Into<String>) -> OpenAICompatibleProvider {
    OpenAICompatibleProvider::new(client, api_key, "meta", tenant_id, "https://api.together.xyz/v1", "llama-3.3-70b-instruct")
}
